packages/adapter-next/src/hooks/slice-template-library-read.ts

```diff
@@ -0,0 +1,46 @@
+import fs from "node:fs";
+import path from "node:path";
+
+import type {
+	SliceTemplateLibraryReadHook,
+	SliceTemplateLibraryReadHookData,
+} from "@slicemachine/plugin-kit";
+
+import type { PluginOptions } from "../types";
+
+import * as CallToAction from "../sliceTemplates/CallToAction";
+
+const initialTemplates = [CallToAction];
+
+export const sliceTemplateLibraryRead: SliceTemplateLibraryReadHook<
+	PluginOptions
+> = async ({
+	templateIds,
+	isTypeScriptProject,
+}: SliceTemplateLibraryReadHookData) => {
+	const templates =
+		templateIds && templateIds.length
+			? initialTemplates.filter((t) => templateIds?.includes(t.model.id))
+			: initialTemplates;
+
+	return {
+		templates: templates.map((t) => {
+			const { mocks, model, createComponentContents, screenshotPaths } = t;
+
+			return {
+				mocks,
+				model,
+				componentContents: createComponentContents(isTypeScriptProject),
+				screenshots: Object.entries(screenshotPaths).reduce(
+					(acc, curr) => ({
+						...acc,
+						[curr[0]]: fs.readFileSync(
```

Comment:
The manager and all adapters use `node:fs/promise` exclusively. In this case, that means changing from `readFileSync` to `readFile`. I understand that adds a bit more code since we need to await the `Promise`s, but it keeps the code async.

Reply:
Done